src/sinks/util/buffer/metrics.rs

```diff
@@ -53,12 +44,10 @@ impl PartialEq for MetricEntry {
         // This differs from a straightforward implementation of `eq` by
         // comparing only the "shape" bits (name, tags, and type) while
         // allowing the contained values to be different.
```

Comment:
Some unit (or docs tests) to cover the behavior which is described here would be nice.

Reply:
This is going to be changing in a follow-up to this work, which is the whole motivation for this change. Metric buffering should *only* be keying on the `MetricSeries` part, but that requires structural changes to the Metrics buffer.